RandLAPACK/drivers/rl_rbki.hh

```diff
@@ -115,488 +115,500 @@ class RBKI : public RBKIalg<T, RNG> {
-    }
-
-    // Convert X_i into an explicit form. It is now stored in X_ev as it should be.
-    lapack::ungqr(m, k, k, X_i, m, tau);
-
-    if(this -> timing) {
-        ungqr_t_stop  = steady_clock::now();
-        ungqr_t_dur   += duration_cast<microseconds>(ungqr_t_stop - ungqr_t_start).count();
-    }
-
-    // Advance odd iteration count.
-    ++iter_od;
-    // Advance iteration count.
-    ++iter;
-
-    // Iterate until in-loop termination criteria is met.
-    while(1) {
-        if(this -> timing)
-            main_loop_t_start = steady_clock::now();
-
-        if (iter % 2 != 0) {
-            if(this -> timing)
-                gemm_A_t_start = steady_clock::now();
-            // Y_i = A' * X_i 
-            blas::gemm(Layout::ColMajor, Op::Trans, Op::NoTrans, n, k, m, 1.0, A, m, X_i, m, 0.0, Y_i, n);
-
-            if(this -> timing) {
-                gemm_A_t_stop = steady_clock::now();
-                gemm_A_t_dur  += duration_cast<microseconds>(gemm_A_t_stop - gemm_A_t_start).count();
-            }
-
-            // Allocate more spece for Y_od
-            curr_X_cols += k;
-            X_ev = ( T * ) realloc(X_ev, m * curr_X_cols * sizeof( T ));
-            // Move the X_i pointer;
-            X_i = &X_ev[m * (curr_X_cols - k)];
-
-            //X_i = &X_i[m * k];
+        ) {
+            linops::GenLinOp<T> A_linop(m, n, A, lda, Layout::ColMajor);
+            return this->call(A_linop, lda, k, U, VT, Sigma, state);
+        }
 
-            if (iter != 1) {
-                // R_i' = Y_i' * Y_od
-                blas::gemm(Layout::ColMajor, Op::Trans, Op::NoTrans, k, iter_ev * k, n, 1.0, Y_i, n, Y_od, n, 0.0, R_i, n);
-                
-                if(this -> timing)
-                    reorth_t_start  = steady_clock::now();                
-                
-                // Y_i = Y_i - Y_od * R_i
-                blas::gemm(Layout::ColMajor, Op::NoTrans, Op::Trans, n, k, iter_ev * k, -1.0, Y_od, n, R_i, n, 1.0, Y_i, n);
+        // RBKI call that accepts sparse matrix.
+        template <RandBLAS::sparse_data::SparseMatrix SpMat>
+        int call(
+            int64_t m,
+            int64_t n,
+            SpMat &A,
+            int64_t lda,
+            int64_t k,
+            T* U,
+            T* VT,
+            T* Sigma,
+            RandBLAS::RNGState<RNG> &state
+        ) {
+            linops::SpLinOp<T, SpMat> A_linop(m, n, A, lda, Layout::ColMajor);
+            return this->call(A_linop, lda, k, U, VT, Sigma, state);
+        }
 
-                // Reorthogonalization
-                blas::gemm(Layout::ColMajor, Op::Trans, Op::NoTrans, k, iter_ev * k, n, 1.0, Y_i, n, Y_od, n, 0.0, Y_orth_buf, k);
-                blas::gemm(Layout::ColMajor, Op::NoTrans, Op::Trans, n, k, iter_ev * k, -1.0, Y_od, n, Y_orth_buf, k, 1.0, Y_i, n);
+        template <RandLAPACK::linops::LinearOperator GLO>
+        int call(
+            GLO& A,
+            int64_t lda,
```

Comment:
``lda`` should not appear as an argument in this function.

Reply:
Fixed.